encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
-
-You will have to crop them beforehand
-or they will return odd graphs like this.
+```py
+from nnedi3_resample import nnedi3_resample
 
-For a change of pace,
-let's look at a good graph.
+upscaled = nnedi3_resample(descaled, width=1920, height=1080)
+```
 
-![Aikatsu Friends! — NCOP (getnative graph)](images/descale_good_graph.png)
+Of course you can still opt for more generic scalers,
+like Spline36 or Lanczos.
 
+```py
+upscaled = core.resize.Spline36(descaled, width=1920, height=1080)
 ```
-Output:
-Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
-Native resolution(s) (best guess): 810p
+
+```py
+upscaled = core.resize.Lanczos(descaled, width=1920, height=1080, filter_param_a=3)
 ```
-The results are very clear.
-There are a couple of deviations,
-but there's a very clear spike
-going down to 810p.
-This is a good result
-for testing out varying kernels.
 
+A couple more extreme examples
+would be using [waifu2x][] or [upscaled_sraa][]
+to upscale it back to 1080p.
+These are far more destructive
+and take much longer,
+but depending on your source,
+might yield overall better results.
 
-## Descaling
+```py
+upscaled = core.caffe.Waifu2x(descaled, noise=-1, scale=2, model=6, cudnn=True)
+upscaled = core.resize.Spline36(upscaled, width=1920, height=1080)
+```
 
-Now it's time to actually start descaling.
-Open up your VapourSynth editor of choice,
-and import the clip:
+```py
+import lvsfunc as lvf
 
-```Py
-src = core.lsmas.LWLibavSource("BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
+upscaled = lvf.upscaled_sraa(descaled, h=1080)
 ```
 
-The next issue is figuring out
-what was used to [upscale](resampling.md#upsampling) the show.
-By default,
-getnative.py checks with Mitchell-Netravali
-(bicubic b=1/3, c=1/3).
-However, it might have also been upscaled
-using other kernels.
-
-Here is a list
-of some common kernels and values.
+[nnedi3_rpow2]: https://github.com/darealshinji/vapoursynth-plugins/blob/master/scripts/nnedi3_rpow2.py
+[nnedi3_resample]: https://github.com/mawen1250/VapourSynth-script/blob/master/nnedi3_resample.py
+[waifu2x]: https://github.com/HomeOfVapourSynthEvolution/VapourSynth-Waifu2x-caffe
+[upscaled_sraa]: https://github.com/Irrational-Encoding-Wizardry/lvsfunc/blob/d4bdfa76f9aa94a49bd038d98e8b28876ab7332c/lvsfunc.py#L515-L582
+
+
+# Handling native 1080p content
+
+When descaling,
+it's important to be wary of native 1080p elements.
+A common example of this is credits,
+which are generally added after upscaling.
+These will require being masked.
+The easiest way to do this
+is by using the re-upscaled clip
+used for checking for the kernel
+and using an expression to get the absolute difference between the two
+and then tweaking the mask further to properly catch everything.
+Then the mask is binarized
+so it doesn't catch every small extra bit
+it happened to catch.
+This should work fine for most of the frame
+(given that you descaled it correctly).
+Here is an example of that:
```

Comment:
Not happy with this paragraph. It's a bit too unprecise imo and worded poorly. On one hand, you describe the steps required, but on the other you leave out important parts of the steps. Like, you say to "tweak" the mask (which at this point is just the difference between two clips) instead of expanding it.